python/pyspark/pandas/tests/test_groupby.py

```diff
@@ -2256,9 +2256,12 @@ def sum_with_acc_frame(x) -> ps.DataFrame[np.float64, np.float64]:
             acc += 1
             return np.sum(x)
 
-        actual = psdf.groupby("d").apply(sum_with_acc_frame).sort_index()
```

Comment:
this reason is:

1, after this PR, dataframe will not be cached since it only contain 1 partition;
2, there is a global sort in `sort_index`, which contains a sampling that will trigger an action. This sampling will cause accumulator be computed twice, this is a already-know issue (see https://issues.apache.org/jira/browse/SPARK-37487)

There maybe a optimization space that convert global sort on single partition to local sort on sigle partition, but I am not sure whether it is worthwhile.

Reply:
Nice